### ISSUE TITLE
Allow overriding the use of locally cached data

### ISSUE THREAD

**Issue by ethanwhite** [weecology/retriever#134]
By default if the Retriever has already downloaded a dataset it will use a locally cached version of the files instead of redownloading them. In some cases this may not be desirable. For example, if the newest year of Breeding Bird Survey data has been released the user may want to rebuild the database with the new data.

Solution: add a `--not-cached` that deletes the locally cached data before proceeding.

**Comment by ShivamNegi** [comment]
I am working on this issue.

**Comment by ShivamNegi** [comment]
I want some help here. I have updated the ```./lib/templates.py``` to incorporate ```--not-cached``` argument. But there are still many scripts which are not using the ```class BasicTextTemplate``` for downloading. These scripts have their own download functions. I can't seem to find any other way than to go in each file and change its download function. Any thought on how to go about incorporating the ```--not-cached``` functionality?

**Comment by ethanwhite** [comment]
Implemented in bc0c2318a854de6c844bace6a28807f54328eae8. Thanks @ShivamNegi!

**Comment by ShivamNegi** [comment]
@ethanwhite after bc0c231. Running ```install``` output is shown below.
```
➜  retriever git:(vertnet-dataset) ✗ retriever install csv vertnet
=> Installing vertnet:
download() got an unexpected keyword argument 'use_cache'
Done!
```
This can be fixed by adding a ```use_cache``` default argument in the ```download``` function of ```vertnet```.

**Comment by ethanwhite** [comment]
Ah yes, thanks for catching that. This will actually cause issues for a number of files:

```
scripts/la_selva_trees.py:    def download(self, engine=None, debug=False):
scripts/gentry_forest_transects.py:    def download(self, engine=None, debug=False):
scripts/breed_bird_survey.py:    def download(self, engine=None, debug=False):
scripts/forest_inventory_analysis.py:    def download(self, engine=None, debug=False):
scripts/prism_climate.py:    def download(self, engine=None, debug=False):
scripts/breed_bird_survey_50stop.py:    def download(self, engine=None, debug=False):
scripts/biomass_allometry_db.py:    def download(self, engine=None, debug=False):
scripts/bioclim.py:    def download(self, engine=None, debug=False):
scripts/plant_life_hist_eu.py:    def download(self, engine=None, debug=False):
scripts/vertnet.py:    def download(self, engine=None, debug=False):
scripts/wood_density.py:    def download(self, engine=None, debug=False):
scripts/intertidal_abund_me.py:    def download(self, engine=None, debug=False):
scripts/npn.py:    def download(self, engine=None, debug=False):
```

@ShivamNegi - do you have time to open another PR with the appropirate fixes?

**Comment by ShivamNegi** [comment]
Yes, I will do that. After some time.

**Comment by ethanwhite** [comment]
Thanks. Much appreciated.

**Comment by ethanwhite** [comment]
Additional fixes made in 3ac063c. Thanks again @ShivamNegi!
